iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -144,29 +162,37 @@ static void sendCompleteCallback(void* context, IOTHUB_MESSAGING_RESULT messagin
 
 static int IoTHubCallback(void* context, const char* data, size_t size)
```

Comment:
Is this always going to work the way we want?  It seems it depends on all messages being delivered on the same callback?  Since we're intentionally going too large for the batch size - never mind distributed system won't promise the service side gets in same chunk size as was sent - I'm not sure we can make this assumption?

Reply:
The way this implementation works is that this `IoTHubCallback` will be called once for each message received by the **test service client**.
So I am not sure what assumption you are bringing up. I'll keep this open until we clarify the ask here.

Reply:
Apologies - I completely misread how this code and the callback generally works.  This is indeed fine.  And it's a reading error on my part - given all the moving pieces I think this is the best (or even only) way to solve this as you've implemented.

Reply:
Thanks :)
I'm glad we double-checked it.